scripts/NGI_pipeline_test.py

```diff
@@ -0,0 +1,238 @@
+        # instrument name
+        instrument = generate_random_instrument()
+        # run id
+        run_id = generate_random_numbers(4)
+        # FC position
+        FC_pos = random.choice(["A", "B"])
+        # FC name
+        FC_name = generate_random_chars(9)
+        # now compose the name
+        run_name = "{}_{}_{}_{}{}".format(date, instrument, run_id, FC_pos, FC_name)
+        # check if this FC exists already
+        if not os.path.exists(run_name):
+            # this FC does not exists, fine
+            new = True
+    return run_name
+
+
+def generate_random_chars(chars=0):
+    return ''.join(random.choice('ABCEFGHILMNOPQRSTUVZ') for i in xrange(chars))
+
+def generate_random_numbers(numbers=0):
+    return ''.join(str(random.randint(0,9)) for i in xrange(numbers))
+
+def generate_random_instrument():
+    instruments = ["ST-E00201", "ST-E00202", "ST-E00203", "ST-E00204", "ST-E00205"] # five instruments
+    return  instruments[random.randint(0,4)]
+
+
+
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser("""This script generates a test suite for NGI-pipeline and piper testing. Needs as input two fastq files (read 1 and read 2) and creates as many FC as specified. The enviorment variables need to point to cahron-dev """)
+    parser.add_argument('--fastq1', help="path to fastq file containing read 1", type=str,  required=True)
+    parser.add_argument('--fastq2', help="path to fastq file containing read 2", type=str,  required=True)
+    parser.add_argument('--FC', help="number of FC to be created", type=int,  required=True)
+    parser.add_argument('--project-name', help="Name of the project that will be simulated (something like J.Doe_16_01)", type=str,
+        default=None)
+    parser.add_argument('--facility', help="facility sequencing the project (stkholm, uppsala)", type=str, default="stockholm",
+        choices=('stockholm, uppsala'))
+    parser.add_argument("-t", "--token", dest="token", default=os.environ.get('CHARON_API_TOKEN'),
+            help="Charon API Token. Will be read from the env variable CHARON_API_TOKEN if not provided")
+    parser.add_argument("-u", "--url", dest="url", default=os.environ.get('CHARON_BASE_URL'),
+            help="Charon base url. Will be read from the env variable CHARON_BASE_URL if not provided")
+    parser.add_argument("--delete", help="delete from charon the project id specified", type=str )
+    parser.add_argument("--symlinks", help="instead of copying files creates symlinks", action='store_true', default=False)
+    args = parser.parse_args()
+
+    if not args.token :
+        print( "No valid token found in arg or in environment. Exiting.")
+        sys.exit(-1)
+    if not args.url:
+        print( "No valid url found in arg or in environment. Exiting.")
+        sys.exit(-1)
+
+    if 'dev' not in args.url:
+        print( "Something tells me that you not using charon-dev.... {}".format(args.url))
+        sys.exit(-1)
```

Comment:
good, I was just thinking that some safeguarding of charon was needed. Normally I would say that hardcoding is bad but in this case it is sort of the point to make it hard to change, so :+1: